### ISSUE TITLE
Not possible to style InspectorControls for an individual block

### ISSUE THREAD

**Issue by iandunn** [WordPress/gutenberg#22198]
### Problem

I need to tweak the CSS for `.block-editor-block-styles__item-preview` on my block, but can't target it without also affecting all other blocks.

There's no `id`/`class` indicating which block the current controls are associated with. Passing a `className` to `InspectorControls`, etc doesn't have an affect.

The same situation could potentially apply to `block-editor-block-card`, and any other elements that Gutenberg automatically adds to the sidebar.

### Potential Solutions
 
* Automatically add a class for the current block to the top level `.components-panel`.
* Allow devs to manually pass in a `className` prop.

**Comment by mtias** [comment]
Can you expand a bit on what do you want to modify within the inspector? We don't want to make classes / IDs be considered a public APIs on these components because they are subject to change as the UI evolves and it'll be tough to ensure compatibility.

**Comment by iandunn** [comment]
I overrode `.block-editor-block-styles__item-preview { align-items: center }` to be `flex-start`, because `center` looks weird for one of my styles.

### Before

<img width="268" alt="Screen Shot 2020-05-08 at 8 30 31 AM" src="https://user-images.githubusercontent.com/484068/81620075-dfad9680-939f-11ea-855f-4f0424431568.png">

### After

<img width="267" alt="Screen Shot 2020-05-08 at 8 30 46 AM" src="https://user-images.githubusercontent.com/484068/81620079-e0dec380-939f-11ea-929e-a2fbe15f00ac.png">

**Comment by mcsf** [comment]
> because `center` looks weird for one of my styles.

Might this be an editor-level improvement to be had?

**Comment by iandunn** [comment]
Do you mean replacing `center` with `flex-start` in `block-styles/styles.scss` itself? That'd WFM in this case.

@youknowriad, did you have any use cases in mind in 622362a6 where `center` worked better?

**Comment by youknowriad** [comment]
Button, I believe we should add a "max-height" to the block style thumbnails here to fix this issue.

**Comment by iandunn** [comment]
🤔 , that might work, although there might not be one value that works for all blocks. In my case, I'd probably use something like `200px`.

I think it'll still need `flex-start`, though, so that the preview shows the top of the content instead of the middle.

**Comment by youknowriad** [comment]
do you already have an "example" set for your block? If it's defined, it will be used for the style variations. You might be able to tweak the example to have the desired preview.

**Comment by iandunn** [comment]
Yeah. The content in it is already paired-down from what real-world content would be, and having `200px` worth of it is important to show the difference between the two styles.

**Comment by vincedubroeucq** [comment]
Hi,
Regardless of specific problems, that could be nice to simply add the blockname to the CSS classes in the inspector controls main wrapper.
Is it a good idea ? Doable ?

**Comment by kassemEzz** [comment]
Hi
I know this is an old issue, but was wondering if a way now exists to add maybe a class to `<InspectorControls>`?
Or maybe someone has found a workaround to style a specific panel title?

thanks

**Comment by kassemEzz** [comment]
> Hi I know this is an old issue, but was wondering if a way now exists to add maybe a class to `<InspectorControls>`? Or maybe someone has found a workaround to style a specific panel title?
> 
> thanks

nvm. It seems `PanelBody` does accept the `className` attribute.
